### PR TITLE
Mr Store 2002

### PR DESCRIPTION
**Pull request by Shiverwarp**

### CODE REVIEW

**Review by horrible-little-slime — DISMISSED** [review]
Does breakfast currently use the catalog? otherwise we need a task to do that, to get your daily credits

File: src/tasks/dailyItems.ts [hunk 1]
```diff
@@ -148,6 +148,21 @@ export const DailyItemTasks: Task[] = [
         }
       },
     },
+    {
+      name: "2002 Mr. Store",
+      // eslint-disable-next-line libram/verify-constants
+      ready: () => have($item`2002 Mr. Store Catalog`),
+      completed: () =>
+        get("availableMrStore2002Credits", 0) > 0 && get("_2002MrStoreCreditsCollected", false),
```
Comment (ajcoppa) [inline src/tasks/dailyItems.ts:156]: You want there to be 0 credits left for this to be completed, right?

```suggestion
        get("availableMrStore2002Credits", 0) === 0 && get("_2002MrStoreCreditsCollected", false),
```

**Comment by ajcoppa** [comment]
> Does breakfast currently use the catalog? otherwise we need a task to do that, to get your daily credits

Looks like mafia already handles this https://github.com/kolmafia/kolmafia/pull/1744

File: src/tasks/dailyItems.ts (hunk 1 above)
Comment (Shiverwarp) [reply src/tasks/dailyItems.ts:156]: Oopsie

**Review by ajcoppa — DISMISSED** [review]

File: src/tasks/dailyItems.ts [hunk 2]
```diff
@@ -148,6 +148,21 @@ export const DailyItemTasks: Task[] = [
...
+    {
+      name: "2002 Mr. Store",
+      // eslint-disable-next-line libram/verify-constants
+      ready: () => have($item`2002 Mr. Store Catalog`),
+      completed: () =>
+        get("availableMrStore2002Credits", 0) === 0 && get("_2002MrStoreCreditsCollected", false),
+      do: (): void => {
+        const bestItem = maxBy(
+          // eslint-disable-next-line libram/verify-constants
+          $items`Spooky VHS Tape, "I survived Y2K" T-Shirt, Letter from Carrie Bradshaw, pro skateboard, Mr. Accessaturday, Meat Butler, Loathing Idol Microphone, Charter: Nellyville, Manual of Secret Door Detection, Flash Liquidizer Ultra Dousing Accessory, Amulet of Perpetual Darkness, Giant black monolith`,
```
Comment (Rinn) [inline src/tasks/dailyItems.ts:160]:
```suggestion
          $items`Spooky VHS Tape, "I survived Y2K" T-Shirt, Letter from Carrie Bradshaw, pro skateboard, Mr. Accessaturday, Meat Butler, Loathing Idol Microphone, Charter: Nellyville, Manual of Secret Door Detection, Flash Liquidizer Ultra Dousing Accessory, Amulet of Perpetual Darkness, Giant black monolith, Crimbo cookie sheet`,
```

**Review by Rinn — COMMENTED** [review]
no crimbo cookie sheet?

File: src/tasks/dailyItems.ts (hunk 2 above)
Comment (Shiverwarp) [reply src/tasks/dailyItems.ts]: Apparently I'm a cookie hater, I used phreds idea for sellsItem to fix it

**Review by s-k-z — CHANGES_REQUESTED** [review]
Need a minimum version requirement increase to at least 27399

**Review by s-k-z — APPROVED** [review]

**Review by Rinn — APPROVED** [review]

**Comment by Veracity0** [comment]
> Need a minimum version requirement increase to at least 27399

Coinmasters with digital currency - credits or dimes or quarters, say - are not new.
Coinmasters that give you more credits by buying things from you are not new.
Coinmasters that give you more currency at rollover which automatically appears in inventory are not new. (SpinMaster Lathe)
Coinmasters that give you more currency at rollover, but only when you "use" an item, rather than automatically appearing when you visit the shop directly, is a brand-new new mechanism.

It took us a number of iterations to figure out all the ramifications. 

27388 added Mr. Store 2002 Coinmaster and tracks available store credits in availableMrStore2002Credits
27392 added _2002MrStoreCreditsCollected, sets if you "use" the item, and uses the item in breakfast
27393 will "use" the item the first time you "visit" the shop, updating both properties
27399 will "use" the item for first purchase, if needed, even if not a visit, updating both properties

At this point, doing any manipulation of the store - Coinmasters frame, "buy", "retrieve_item", etc. - will "use" the item the first time (to collect daily credits) and will parse the shop to see your current balance.

However, simply "using" the item - Item Manager, "use" command or function, breakfast - will successfully collect daily credits, but will not parse the shop to update your balance. Additionally, "using" the item in the Relay Browser will visit the shop - and update balances - but will not set _2002MrStoreCreditsCollected,

27400 fixes all those things.

Hopefully, that's all (knock on wood).

But if you want breakfast to not only collect new credits but also update the available balance, 27400 is what you need.

**Pull request merged**